src/Sulu/Bundle/AdminBundle/Resources/js/containers/List/List.js

```diff
@@ -502,12 +518,17 @@ class List extends React.Component<Props> {
         return (
             <div className={listStyles.listContainer}>
                 {header}
-                {!store.schemaLoading && (searchable || this.currentAdapter.hasColumnOptions || adapters.length > 1) &&
+                {!schemaLoading && (searchable || adapters.length > 1 || filterable || hasColumnOptions) &&
                     <div className={toolbarClass}>
                         {searchable &&
                             <Search onSearch={this.handleSearch} value={store.searchTerm.get()} />
                         }
-                        {this.currentAdapter.hasColumnOptions &&
+                        <FieldFilter
```

Comment:
It will, and that's on purpose, because the filters will take all the space between the search and column options button. If we remove it, if no filters are applied, the search button will move the right, causing an inconsistency in the UI.

Reply:
okay it does not feel intuitive to me that the `List` container relies on the `FieldFilter` component to expand. especially because the `FieldFilter` component has no internal/own reason for doing this.

Reply:
Well, internally the `FieldFilter` component behaves exactly the same, and does not take care of this. The component that needs this, is the `List` container, therefore I think it is find the way it currently is.

The alternative that comes to my mind is to have a more complex DOM setup, instead of just relying on flex-box. But I like that option less than the current one, since I prefer to rely on CSS as much as possible, in order to keep the markup simple :slightly_smiling_face:

Reply:
the `fieldFilter.scss` contains `flex-grow: 1;` even though there is no reason for this in the `FieldFilter.js`. the only reason the `flex-grow: 1;` is there is because the `List` container (which the `FieldFilter` should not know about) needs it. 

i dont think this is intuitive and would probably remove it when i am refactoring something there. furthermore, there is not test case that would detect this as removing the style would not change anything visible in our snapshot test.

but i understand that you dont want to make the dom more complex just because of this. the correct solution would probably to pass an additional class prop from the `List` to the `FieldFilter`.  but thats quite some work for a single style property so i can live with the current solution too, just dont like it 😉

Reply:
That's also an opinion I don't share :see_no_evil: The `FieldFilter` is part of the `List`, and not used anywhere else, it is even in the same Folder. Since it is part of the list, it is reasonable that it can make some assumptions about it

There are many other CSS properties that require certain stuff to be set on its parent, otherwise they won't work as expected. And if we can't rely on that within components that belong together, we are making our lives much harder than they need to be.

Apart from that non of our tests currently check for CSS changes, no matter how we structure this. Just because I see a class in the snapshot, it does not mean that it is rendering correctly. That's why I would love to introduce some screenshot tests for critical parts of the application like the `Grid`.